src/python/pants/backend/awslambda/common/awslambda_common_rules.py

```diff
@@ -64,15 +59,19 @@ async def create_awslambda(
         for field_set in targets_to_valid_field_sets.field_sets
     )
     merged_digest = await Get(Digest, MergeDigests(awslambda.digest for awslambda in awslambdas))
-    result = workspace.materialize_directory(
-        DirectoryToMaterialize(merged_digest, path_prefix=str(distdir.relpath))
-    )
+    workspace.write_digest(merged_digest, path_prefix=str(distdir.relpath))
     with options.line_oriented(console) as print_stdout:
-        for awslambda, path in zip(awslambdas, result.output_paths):
```

Comment:
This `zip()` was not safe. There was no guarantee that `awslambdas` had the same order as `result.output_paths`.